nni/nas/nn/pytorch/repeat.py

```diff
@@ -61,80 +67,154 @@ class Repeat(Mutable):
                 warnings.warn(
-                    'In repeat, `depth` is already a ValueChoice, but `label` is still set. It will be ignored.',
+                    'In repeat, `depth` is already a mutable, but `label` is still set. It will be ignored.',
                     RuntimeWarning
                 )
-            self.depth_choice: Union[int, ChoiceOf[int]] = depth
-            all_values = list(self.depth_choice.all_options())
+
+            self.depth_choice: Union[int, MutableExpression] = cast(Union[int, MutableExpression], depth)
+            all_values = _all_finite_integers(depth)
             self.min_depth = min(all_values)
             self.max_depth = max(all_values)
 
-            if isinstance(depth, ValueChoice):
-                self._label = depth.label  # if a leaf node
-
         elif isinstance(depth, tuple):
             self.min_depth = depth if isinstance(depth, int) else depth[0]
             self.max_depth = depth if isinstance(depth, int) else depth[1]
-            self.depth_choice: Union[int, ChoiceOf[int]] = ValueChoice(list(range(self.min_depth, self.max_depth + 1)), label=label)
+            self.depth_choice: Union[int, MutableExpression] = Categorical(list(range(self.min_depth, self.max_depth + 1)), label=label)
             self._label = self.depth_choice.label
 
         elif isinstance(depth, int):
             self.min_depth = self.max_depth = depth
-            self.depth_choice: Union[int, ChoiceOf[int]] = depth
+            self.depth_choice: Union[int, MutableExpression] = depth
         else:
             raise TypeError(f'Unsupported "depth" type: {type(depth)}')
         assert self.max_depth >= self.min_depth >= 0 and self.max_depth >= 1, f'Depth of {self.min_depth} to {self.max_depth} is invalid.'
+
+        if isinstance(self.depth_choice, Mutable):
+            self.add_mutable(self.depth_choice)
+            self._dry_run_depth = ensure_frozen(self.depth_choice)
+        else:
+            self._dry_run_depth = self.depth_choice
+
+        if isinstance(blocks, nn.ModuleList):
+            _logger.warning('Using ModuleList as blocks will make the whole module list be treated as one block, '
+                            'and it will be deep-copied. This might be not intended in most cases. '
+                            'Consider using a pure python list of modules if you want to specify a sequence of blocks.')
+
         self.blocks = nn.ModuleList(self._replicate_and_instantiate(blocks, self.max_depth))
 
+    @torch.jit.unused
     @property
     def label(self) -> Optional[str]:
-        return self._label
+        if isinstance(self.depth_choice, LabeledMutable):
+            return self.depth_choice.label
+        return None
 
     def forward(self, x):
-        for block in self.blocks:
-            x = block(x)
+        for i, block in enumerate(self.blocks):
+            if i < self._dry_run_depth:
+                x = block(x)
+            # Make JIT happy
+            # else:
```

Comment:
?

Reply:
Should have an `else` here. But we can't because otherwise pytorch.jit will complain.

This is again for the (useless) graph converter.